airbyte-cdk/bulk/toolkits/load-object-storage/src/main/kotlin/io/airbyte/cdk/load/file/object_storage/ObjectStoragePathFactory.kt

```diff
@@ -321,11 +331,13 @@ class ObjectStoragePathFactory(
         return Regex.escapeReplacement(input).replace(macroPattern.toRegex()) {
             val variable = it.groupValues[1]
             val pattern = variableToPattern[variable]
-            if (pattern != null) {
+            if (pattern == null) {
+                variable
+            } else if (pattern.isBlank()) {
+                ""
```

Comment:
Nit: maybe leave a comment here about why this branching exists. I don't entirely follow myself.

Reply:
```
                // This should only happen in the case of a blank namespace.
                // This is to avoid inserting `()` and then trying to match
                // `()/($streamName)` against `$streamName`.
```